src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs

```diff
@@ -7468,6 +7468,277 @@ public void SyntaxOffset_OutVarInMethod()
 ");
         }
 
+        [Fact]
+        public void SyntaxOffset_OutVarInInitializers_01()
```

Comment:
>public void SyntaxOffset_OutVarInInitializers_01( [](start = 8, length = 49)

We need similar tests with expression variables that are closed over in lambdas.

Eg.
```C#
class C { C() : base(G(out var z) && F(() => z)) { } }
```

```C#
class C { bool x = G(out var z) && F(() => z)); }
```

In these cases we need to validate that the closure information emitted to the PDB is correct.

Reply:
See e.g. [https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs#L224-L232](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs#L224-L232), where these kind of tests live.


---
In reply to: [174562033](https://github.com/dotnet/roslyn/pull/25441#discussion_r174562033) [](ancestors = 174562033)

Reply:
Also [https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs#L2356](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs#L2356) verify that the members generated for the lambda and closures are correctly reused when edited.

---
In reply to: [174562799](https://github.com/dotnet/roslyn/pull/25441#discussion_r174562799) [](ancestors = 174562799,174562033)